lib/slayer/result.rb

```diff
@@ -8,16 +8,18 @@ def initialize(value, status, message)
       @message = message
     end
 
-    def success?
-      !failure?
+    def passed?
+      !failed?
     end
+    alias success? passed?
```

Comment:
Also need an alias for `ok?`. I still have mixed feelings about the naming here.

Reply:
I don't want to introduce more aliases — I want to settle on method names and provide only those + the deprecated backwards compatibility layer.

I can be convinced that `ok?` is better than `passed?` — but I don't want to use both.

Reply:
I just had a flash of insight here. Lets make the backward compatibility layer a separate module you have to import and explicitly opt into. Ruby makes this particularly clean.

```rb
# imports the "blessed" API
require 'slayer' 

# imports compatibility layer
require 'slayer/compatibility'
```

I think we could even add `slayer/compatibility/v0.4.0` and stuff.